h2o-core/src/main/java/water/rapids/ast/prims/reducers/AstTopN.java

```diff
@@ -0,0 +1,167 @@
+package water.rapids.ast.prims.reducers;
+
+import water.MRTask;
+import water.fvec.C8Chunk;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.rapids.Env;
+import water.rapids.ast.AstPrimitive;
+import water.rapids.ast.AstRoot;
+import water.rapids.vals.ValFrame;
+
+import java.util.PriorityQueue;
+
+import static java.lang.StrictMath.min;
+
+public class AstTopN extends AstPrimitive {
+  @Override
+  public String[] args() {
+    return new String[]{"frame", "col", "nPercent", "getBottomN"};
+  }
+
+  @Override
+  public String str() {
+    return "topn";
+  }
+
+  @Override
+  public int nargs() {
+    return 1+4;
+  } // function name plus 4 arguments.
+
+  @Override
+  public String example() {
+    return "(topn frame col nPercent getBottomN)";
+  }
+
+  @Override
+  public String description() {
+    return "Return the top N percent rows for a numerical column as a frame with two columns.  The first column " +
+            "will contain the original row indices of the chosen values.  The second column contains the top N row" +
+            "values.  If getBottomN is 1, we will return the bottom N percent.  If getBottomN is 0, we will return" +
+            "the top N percent of rows";
+  }
+
+  @Override
+  public ValFrame apply(Env env, Env.StackHelp stk, AstRoot[] asts) { // implementation with PriorityQueue
+    Frame frOriginal = stk.track(asts[1].exec(env)).getFrame(); // get the 2nd argument and convert it to a Frame
+    int colIndex = (int) stk.track(asts[2].exec(env)).getNum();     // column index of interest
+    double nPercent = stk.track(asts[3].exec(env)).getNum();        //  top or bottom percentage of row to return
+    int getBottomN = (int) stk.track(asts[4].exec(env)).getNum();   // 0, return top, 1 return bottom percentage
+    int totColumns = frOriginal.numCols();
+    long numRows = Math.round(nPercent*0.01*frOriginal.numRows()); // number of rows to return
+
+    // check for valid input parameter values
+    assert ((nPercent>0) && (nPercent<=100)); // make sure percent is between 0 and 100.
```

Comment:
Ehh? Is this intention to do user argument checking? Then you should not use assertion which is dev construct only. 

Moreover, I am not aware of any check upper in the stack which would check for values constrains. 
We can of course argue that Rapids are internal language generated by us, however, it is expose via REST API, so anybody can send any rapids and if assertions are disabled it will fail on some random place later in code.